File: tests/testdata/js/ops.js
function mathOps(a, b) {
    const x = a + b;
    const y = x * 2;
    return y - 1;
}

function bitwiseOps(a, b) {
    const x = a | b;
    return x & 1;
}

function comparisons(a, b) {
    if (a === b) {
        return a;
    }
    return b;
}

function logicalOps(a, b, c) {
    return a && b || !c;
}

function conditionals(x) {
    if (x > 0) {
        return x;
    } else if (x < 0) {
        return -x;
    }
    return 0;
}

function callsZero(a) {
    return a;
}

function callsOne() {
    foo();
    return 1;
}

function callsTwo() {
    foo();
    bar();
}

function callsThree() {
    foo();
    bar();
    baz();
}

function withException() {
    try {
        foo();
    } catch (e) {
        bar();
    } finally {
        baz();
    }
}

function withArrow(items) {
    return items.map(x => x + 1);
}

class MyClass {
    method(x) {
        return x * 2;
    }
}

function boilerplate(data) {
    const x = data;
    const y = x;
    const z = y;
    return z;
}

File: tests/testdata/go/ops.go
package ops

func mathOps(a, b int) int {
	x := a + b
	y := x * 2
	return y - 1
}

func bitwiseOps(a, b int) int {
	x := a | b
	return x & 1
}

func comparisons(a, b int) bool {
	return a == b
}

func logicalOps(a, b, c bool) bool {
	return a && b || !c
}

func conditionals(x int) int {
	if x > 0 {
		return x
	} else if x < 0 {
		return -x
	}
	return 0
}

func callsZero(a int) int {
	return a
}

func callsOne() {
	foo()
}

func callsTwo() {
	foo()
	bar()
}

func callsThree() {
	foo()
	bar()
	baz()
}

type MyStruct struct{}

func (s MyStruct) method(x int) int {
	return x * 2
}

func boilerplate(data int) int {
	x := data
	y := x
	z := y
	return z
}
